src/indexer.c

```diff
@@ -151,6 +151,12 @@ int git_indexer_new(
 	if (fd != -1)
 		p_close(fd);
 
+	if (git_buf_is_allocated(&tmp_path))
```

Comment:
`git_buf_len(&tmp_path)` is correct here.

`is_allocated` literally looks at whether we have `malloc`ed the structure, not whether it has contents.  These are orthogonal concepts:  the buffer may point to static memory and thus be nonempty but not allocated, similarly, we could allocate a buffer, set contents into it, then clear the contents, but it would still be allocated.